### PR TITLE
Android: if sent from the background, the ads notifications should not make sound and pop up on the screen.Issue #2419.

### PR DESCRIPTION
**Pull request by gdregalo**
## Submitter Checklist:
Fixes:  https://github.com/brave/browser-android-tabs/issues/2419
Works with: https://github.com/brave/browser-android-tabs/pull/2480
Read comments in: https://github.com/brave/browser-android-tabs/pull/2480

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_ads/browser/notification_helper_android.cc [hunk 1]
```diff
@@ -73,13 +72,14 @@ NotificationHelper* NotificationHelper::GetInstanceImpl() {
 
 ///////////////////////////////////////////////////////////////////////////////
 
-bool NotificationHelperAndroid::IsBraveAdsNotificationChannelEnabled() const {
+bool NotificationHelperAndroid::IsBraveAdsNotificationChannelEnabled(bool foregroundChannel) const {
```
Comment (tmancey) [inline components/brave_ads/browser/notification_helper_android.cc]: Can we please change to snake_case

File: components/brave_ads/browser/notification_helper_android.cc [hunk 2]
```diff
@@ -37,14 +37,13 @@ bool NotificationHelperAndroid::ShouldShowNotifications() {
...
 
-  bool is_notification_channel_enabled = IsBraveAdsNotificationChannelEnabled();
-
-  auto is_foreground = BackgroundHelper::GetInstance()->IsForeground();
-  auto should_show_notifications =
-      CanShowBackgroundNotifications() || is_foreground;
-
-  return is_notifications_enabled && is_notification_channel_enabled &&
-      should_show_notifications;
+  bool is_foreground = BackgroundHelper::GetInstance()->IsForeground();
```
Comment (tmancey) [inline components/brave_ads/browser/notification_helper_android.cc]: This business logic should be moved to `AdsImpl::CheckReadyAdServe`, i.e. see `FailedToServeAd("Not in foreground");) {`

File: components/brave_ads/browser/notification_helper_android.cc [hunk 3]
```diff
@@ -37,14 +37,13 @@ bool NotificationHelperAndroid::ShouldShowNotifications() {
...
+  bool is_notification_channel_enabled = IsBraveAdsNotificationChannelEnabled(is_foreground);
+  bool result = is_notifications_enabled && is_notification_channel_enabled;
+  if (!is_foreground) {
+    result = result && CanShowBackgroundNotifications();
```
Comment (tmancey) [inline components/brave_ads/browser/notification_helper_android.cc]: See above comment as `CanShowBackgroundNotifications` business logic should also be moved to `AdsImpl::CheckReadyAdServe`

File: components/brave_ads/browser/notification_helper_android.cc [hunk 4]
```diff
@@ -73,13 +72,14 @@ NotificationHelper* NotificationHelper::GetInstanceImpl() {
...
 
-bool NotificationHelperAndroid::IsBraveAdsNotificationChannelEnabled() const {
+bool NotificationHelperAndroid::IsBraveAdsNotificationChannelEnabled(bool foregroundChannel) const {
   if (GetOperatingSystemVersion() < kMinimumVersionForNotificationChannels) {
     return true;
   }
 
   JNIEnv* env = base::android::AttachCurrentThread();
-  auto j_channel_id = Java_BraveAds_getBraveAdsChannelId(env);
+  auto j_channel_id = (foregroundChannel)? Java_BraveAds_getBraveAdsChannelId(env):
```
Comment (tmancey) [inline components/brave_ads/browser/notification_helper_android.cc]: nit: space before `?`

File: components/brave_ads/browser/notification_helper_android.cc (hunk 1 above)
Comment (tmancey) [inline components/brave_ads/browser/notification_helper_android.cc]: `const bool`

File: components/brave_ads/browser/notification_helper_android.cc [hunk 5]
```diff
@@ -37,14 +37,13 @@ bool NotificationHelperAndroid::ShouldShowNotifications() {
...
-  bool is_notification_channel_enabled = IsBraveAdsNotificationChannelEnabled();
-
-  auto is_foreground = BackgroundHelper::GetInstance()->IsForeground();
-  auto should_show_notifications =
-      CanShowBackgroundNotifications() || is_foreground;
-
-  return is_notifications_enabled && is_notification_channel_enabled &&
-      should_show_notifications;
+  bool is_foreground = BackgroundHelper::GetInstance()->IsForeground();
+  bool is_notification_channel_enabled = IsBraveAdsNotificationChannelEnabled(is_foreground);
```
Comment (tmancey) [inline components/brave_ads/browser/notification_helper_android.cc]: nit: `const`

File: components/brave_ads/browser/notification_helper_android.cc [hunk 6]
```diff
@@ -37,14 +37,13 @@ bool NotificationHelperAndroid::ShouldShowNotifications() {
...
+  bool result = is_notifications_enabled && is_notification_channel_enabled;
```
Comment (tmancey) [inline components/brave_ads/browser/notification_helper_android.cc]: nit: `const`

File: components/brave_ads/browser/notification_helper_android.cc (hunk 1 above)
Comment (tmancey) [reply components/brave_ads/browser/notification_helper_android.cc]: nit: > 80 line length, can you please run changes through the linter

File: components/brave_ads/browser/notification_helper_android.h [hunk 7]
```diff
@@ -24,7 +24,9 @@ class NotificationHelperAndroid
   NotificationHelperAndroid();
   ~NotificationHelperAndroid() override;
 
-  bool IsBraveAdsNotificationChannelEnabled() const;
+
+  //foregroundChannel: foreground or background channel
```
Comment (tmancey) [inline components/brave_ads/browser/notification_helper_android.h]: nit: Use Chromium style comments, i.e. `|` around variable/function names, i.e. `|foreground_channel|`

File: components/brave_ads/browser/notification_helper_android.cc (hunk 1 above)
Comment (gdregalo) [reply components/brave_ads/browser/notification_helper_android.cc]: done

File: components/brave_ads/browser/notification_helper_android.cc (hunk 4 above)
Comment (gdregalo) [reply components/brave_ads/browser/notification_helper_android.cc]: done

File: components/brave_ads/browser/notification_helper_android.cc (hunk 1 above)
Comment (gdregalo) [reply components/brave_ads/browser/notification_helper_android.cc]: pass by value: doesn't have to be `const`

File: components/brave_ads/browser/notification_helper_android.cc (hunk 5 above)
Comment (gdregalo) [reply components/brave_ads/browser/notification_helper_android.cc]: local temp value doesn't have to be `const`

File: components/brave_ads/browser/notification_helper_android.cc (hunk 6 above)
Comment (gdregalo) [reply components/brave_ads/browser/notification_helper_android.cc]: see above comment

File: components/brave_ads/browser/notification_helper_android.h (hunk 7 above)
Comment (gdregalo) [reply components/brave_ads/browser/notification_helper_android.h]: done

File: components/brave_ads/browser/notification_helper_android.cc (hunk 5 above)
Comment (tmancey) [reply components/brave_ads/browser/notification_helper_android.cc]: Raised nit for consistency with ads

File: components/brave_ads/browser/notification_helper_android.cc (hunk 1 above)
Comment (tmancey) [reply components/brave_ads/browser/notification_helper_android.cc]: See below

File: components/brave_ads/browser/notification_helper_android.cc (hunk 2 above)
Comment (tmancey) [reply components/brave_ads/browser/notification_helper_android.cc]: I will raise a ticket to move this business logic as discussed

File: components/brave_ads/browser/notification_helper_android.cc (hunk 3 above)
Comment (tmancey) [reply components/brave_ads/browser/notification_helper_android.cc]: I will raise a ticket to move this business logic as discussed

**Comment by tmancey** [comment]
@gdregalo CI is failing, please check if linter failed

**Pull request merged**